zombies.js

```diff
@@ -79,6 +105,161 @@
  * @property {method} getMaxHealth         Returns private variable `maxHealth`.
  */
 
+class Player {
+  constructor(name, health, strength, speed) {
+    this._pack = [];
+    this._maxHealth = health;
+    this._name = name;
+    this._health = health;
+    this._strength = strength;
+    this._speed = speed;
+    this._isAlive = true;
+    this._equipped = false;
+  }
+
+  getPack() {
+    return this._pack;
+  }
+
+  getMaxHealth() {
+    return this._maxHealth;
+  }
+
+  get name() {
+    return this._name;
+  }
+
+  get health() {
+    return this._health;
+  }
+
+  set health(energy) {
+    this._health = energy;
+  }
+
+  get strength() {
+    return this._strength;
+  }
+
+  get speed() {
+    return this._speed;
+  }
+
+  get isAlive() {
+    return this._isAlive;
+  }
+
+  get equipped() {
+    return this._equipped;
+  }
+
+  set equipped(item) {
+    this._equipped = item;
+  }
+
+  checkPack() {
+    let pack = this.getPack();
+    if (pack.length === 0) {
+      console.log(this.name + " has nothing in their pack.");
+    }
+    for (let i = 0; i < pack.length; i++) {
+      console.log(this.name + " has " + pack[i].name + " stored.");
+    }
+  }
+
+  takeItem(item) {
+    if (this.getPack().length < 3) {
+      this.getPack().push(item);
+      console.log(item.name + " has been added to " + this.name + "'s pack.");
+      return true;
+    } else {
+      console.log(this.name + " already has 3 items in their pack.");
+      return false;
+    }
+  }
+
+  discardItem(item) {
+    if (this.getPack().indexOf(item) !== -1) {
```

Comment:
You could pull out `this.getPack().indexOf(item)` into a variable here so that it doesn't run twice.